src/aleph/handlers/forget.py

```diff
@@ -119,15 +166,46 @@ async def forget_if_allowed(target_message: Dict, forget_message: ForgetMessage)
         "item_content": None,
         "forgotten_by": [forget_message.item_hash],
     }
-    await Message.collection.update_many(filter={"item_hash": target_hash}, update={"$set": updates})
+    await Message.collection.update_many(
+        filter={"item_hash": target_hash}, update={"$set": updates}
+    )
 
     # TODO QUESTION: Should the removal be added to the CappedMessage collection for websocket
     #  updates ? Forget messages should already be published there, but the logic to validate
     #  them could be centralized here.
 
-    if target_message.get("type") == MessageType.store:
-        store_content = StoreContent(**target_message["content"])
-        await garbage_collect(store_content.item_hash, store_content.item_type)
+    if target_info.type == MessageType.store:
+        if (
+            target_info.content_item_type is None
+            or target_info.content_item_hash is None
+        ):
+            raise ValueError(
+                f"Could not garbage collect content linked to STORE message {target_hash}."
```

Comment:
Can you make this message more explicit ? Something like:
```suggestion
                f"Information missing, could not garbage collect content linked to STORE message {target_hash}."
```